src/Workspaces/Core/Portable/Shared/Utilities/ProducerConsumer.cs

```diff
@@ -83,6 +79,29 @@ internal static class ProducerConsumer<TItem>
         Func<IAsyncEnumerable<TItem>, TArgs, CancellationToken, Task> consumeItems,
         TArgs args,
         CancellationToken cancellationToken)
+    {
+        // Bridge to sibling helper that takes a consumeItems that returns a value.
+        return RunAsync(
+            options,
+            produceItems: static (callback, args, cancellationToken) => args.produceItems(callback, args.args, cancellationToken),
```

Comment:
>static (callback, args, cancellationToken) => args.produceItems(callback, args.args, cancellationToken), [](http://example.com/codeflow?start=26&length=104)

Early, so coffee might not have hit, but could this just be the following instead?

produceItems: produceItems,

Reply:
Nope. Notice the `args.args` part.  The two delegates have definite different shapes.